### PR TITLE
Profile zkApp transactions

### PR DESCRIPTION
**Pull request by deepthiskumar**
Generate zkApp transactions with different permutations of proof and non-proof updates

Checklist:

- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

* Closes #0000

### CODE REVIEW

**Comment by deepthiskumar** [comment]
!ci-build-me

File: src/app/test_executive/zkapps_timing.ml [hunk 1]
```diff
@@ -182,15 +180,15 @@ module Make (Inputs : Intf.Test.Inputs_intf) = struct
           "Invalid transfer, timed account"
       in
       let sender_keypair = timed_account_keypair in
-      let receiver_key =
-        keypair.public_key |> Signature_lib.Public_key.compress
+      let receiver =
+        keypairkeypair.public_key |> Signature_lib.Public_key.compress
```
Comment (psteckler) [inline src/app/test_executive/zkapps_timing.ml:184]: typo?

File: src/lib/transaction_snark/transaction_snark.ml [hunk 2]
```diff
@@ -4357,9 +4361,26 @@ module Make_str (A : Wire_types.Concrete) = struct
       ( `VK (With_hash.of_data ~hash_data:Zkapp_account.digest_vk vk)
       , `Prover trivial_prover )
 
-    let create_zkapp_command
+    let rec interleave left right dir acc =
+      if List.length left = 0 && List.length right = 0 then List.rev acc
```
Comment (psteckler) [inline src/lib/transaction_snark/transaction_snark.ml:4365]: `List.is_empty` is faster

Comment (psteckler) [reply src/lib/transaction_snark/transaction_snark.ml:4365]: Or, better, pattern-match on `[],[]`

File: src/app/cli/src/init/transaction_snark_profiler.ml [hunk 3]
```diff
@@ -508,11 +908,28 @@ let command =
...
+       flag "--max-num-updates" ~aliases:[ "-max-num-updates" ]
+         ~doc:
+           "Maximum number of account updates per transaction (excluding the \
+            fee payer). Default:6"
+         (optional int)
+     and min_num_updates =
+       flag "--min-num-updates" ~aliases:[ "-min-num-updates" ]
+         ~doc:
+           "Minimum number of account updates per transaction (excluding the \
+            fee payer). It should be at least 1. Default:1 "
```
Comment (psteckler) [inline src/app/cli/src/init/transaction_snark_profiler.ml:921]: `Minimum: 1` is less verbose

**Review by psteckler — APPROVED** [review]
Some nits, looks good otherwise

**Comment by deepthiskumar** [comment]
!ci-build-me

**Comment by deepthiskumar** [comment]
!ci-build-me

**Pull request merged**